godot-core/src/builtin/vector3.rs

```diff
@@ -306,6 +306,11 @@ impl Vector3 {
             snapped(self.z, step.z),
         )
     }
+
+    /// Returns this vector rotated around `axis` by `angle` radians. `axis` must be normalized.
+    pub fn rotated(self, axis: Vector3, angle: real) -> Self {
```

Comment:
```suggestion
    pub fn rotated(self, axis: Self, angle: real) -> Self {
```

When looking at the other `fn`s, `Self` is used in favor of `Vector3`

Reply:
Also, could you assert the normalization?

Reply:
Note that this normalization is required by the `fn` on `Basis` as well, it stems from inside Glam. Glam asserts this if the relevant feature is enabled, however we only enable the debug_glam_assert feature. So if we assert it here, we should assert it in the other relevant places as well, to not create inconsistent behavior.

Reply:
This is the particular location for mat3 in glam source: https://docs.rs/glam/latest/src/glam/f32/mat3.rs.html#190